<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<HTML><HEAD><TITLE>Affymetrix BPMAP File Format</TITLE>
<META http-equiv=Content-Type content="text/html; charset=windows-1252">
<META 
content="Affymetrix, GeneChip, GeneChip Array, Netaffx, NetAffx Analysis Center, Affymetrics, Affimetrix, Affimetrics, " 
name=keywords>
<META 
content="Affymetrix is dedicated to developing state-of-the-art technology for acquiring, analyzing, and managing complex genetic information for use in biomedical research. Affymetrix sells GeneChip® brand arrays." 
name=description>
<SCRIPT language=JavaScript>
</SCRIPT>

<SCRIPT language=JavaScript src="files_files/main.js" 
type=text/javascript></SCRIPT>
<LINK href="files_files/style.css" type=text/css rel=stylesheet><LINK 
href="/favicon.ico" rel=icon>
<META content="Microsoft FrontPage 6.0" name=GENERATOR>
<style>
<!--
table.MsoTableClassic4
	{border-top:1.5pt solid black;
	border-left:1.0pt solid black;
	border-bottom:1.5pt solid black;
	border-right:1.0pt solid black;
	font-size:10.0pt;
	font-family:"Times New Roman";
	}
-->
</style>
</HEAD>
<BODY onresize=fixIt() vLink=#003366 aLink=#003366 link=#003366 bgColor=#ffffff 
leftMargin=0 topMargin=0 onunload=window.focus() marginheight="0" 
marginwidth="0">
<!-- Top links and Search plus noindex tags for atomz--> <NOINDEX> <TABLE cellSpacing=0 cellPadding=0 border=0> 
<TBODY> <TR> <TD vAlign=center align=left>&nbsp;</TD></TR> </TBODY> </TABLE><TABLE cellSpacing=0 cellPadding=0 border=0 width="601"> 
<TBODY> <TR> <TD vAlign=top align=left><IMG height=54 alt=Affymetrix src="files_files/affymetrix_logo.gif" width=128 border=0></TD><TD></TD><TD vAlign=center align=left class=pageheader><p>AFFYMETRIX 
DEVELOPER NETWORK</p></TD></TR> </TBODY> </TABLE></NOINDEX><TABLE CELLSPACING=0 CELLPADDING=0 WIDTH=772 BORDER=0> 
<TBODY> <TR> <TD VALIGN=top ALIGN=left WIDTH=772 BGCOLOR=#336699 HEIGHT="2"><IMG height=3 alt="" 
			src="files_files/spacer.gif" width=1 border=0></TD></TR> 
</TBODY> </TABLE><TABLE cellSpacing=0 cellPadding=5 width=409 
      background=files_files/network_background.gif border=0> <TBODY> <TR> <TD vAlign=top align=left> 
<TABLE cellSpacing=0 cellPadding=0 
            background=files_files/spacer.gif border=0> <TBODY> <TR><TD vAlign=top align=left width=11" ><IMG HEIGHT=1 ALT="" 
                        SRC="files_files/spacer.gif" WIDTH=11 BORDER=0></TD><TD vAlign=top align=left width="100%"><IMG height=41 alt="" 
                  src="files_files/spacer.gif" width=1 border=0></TD></TR> <TR> 
<TD vAlign=top align=left width=11" ><IMG HEIGHT=1 ALT="" 
                        SRC="files_files/spacer.gif" WIDTH=11 BORDER=0></TD><TD class=intro vAlign=top align=left> 
<P><span class="header2">Affymetrix® TPMAP File Format</span></P></TD></TR></TBODY></TABLE></TD></TR></TBODY></TABLE>&nbsp;<TABLE cellSpacing=0 cellPadding=5 border=0> 
<TBODY> <TR> <TD vAlign=top align=left> <TABLE cellSpacing=0 cellPadding=0 border=0> 
<TBODY> <TR> <TD vAlign=top align=left> 
  <TABLE cellSpacing=0 cellPadding=0 border=0 > 
<TBODY> 
	<tr>
	<TD vAlign=top align=right style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none" width="11">
		<IMG height=11 alt="" src="files_files/bullet2.gif" width=11 border=0></TD>
	<TD vAlign=top align=left style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none" width="8">
		<IMG height=1 alt="" src="files_files/spacer.gif" width=5 border=0></TD>
	<TD class=header3 vAlign=center align=left  style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none">
	<a name="BPMAP_FILE"></a>TPMAP FILE</TD>
	</tr>
	<tr>
	<TD vAlign=top align=left colSpan=3 style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none" >
			<IMG height=10 alt="" src="files_files/spacer.gif" width=1 border=0></TD>
	</tr>
	<tr>
	<TD vAlign=top align=left style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none" width="11">
		<IMG height=1 alt="" src="files_files/spacer.gif" width=11 border=0></TD>
	<TD vAlign=top align=left style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none" width="8">
		<IMG height=1 alt="" src="files_files/spacer.gif" width=5 border=0></P><P>&nbsp;</P></TD>
	<TD class=content vAlign=top align=left  style="BORDER-TOP-STYLE: none; BORDER-RIGHT-STYLE: none; BORDER-LEFT-STYLE: none; BORDER-BOTTOM-STYLE: none">
		<span class="header4">Description</span><BR> The TPMAP (text probe map) 
		Version 3 file format is used to store the relationship between (PM,MM) 
		probe pairs or PM probes (PM only array) and positions on a set of 
		sequences of interest.<p>Each sequence of interest must be identified by a &quot;Sequence Group Name&quot; 
		(e.g., Species or organism), &quot;Version&quot; (e.g., build of genome database) 
		and &quot;Target sequence name&quot; (e.g., chromosome).</p>
		<P><span class="header4">Format</span><BR>
		The format is plain text.</P>
		<P>Probes in the same Group Name, Version and Target sequences occur in 
		blocks with all entries having the same target sequence name. The first 
		such block must be headed by two lines giving Group Name and Version 
		(any successive blocks without these pairs will inherit them):</P>
		<P>#seq_group_name group_name<br>
		#version version<br>
		<br>
		and then optional tag/value pairs (see below). Fields are whitespace 
		delimited and fields after the first two can be present and will be 
		ignored.</P>
		<P>There is a row corresponding to every instance of a (PM,MM) probe pair 
		or a PM probe aligning to a target sequence. Probe sequence is the 
		synthesis sequence of the probe on the chip in 5'-3' orientation. It is 
		the reverse complement of the target sequence.</P>
		<P>Each row consists of the following 6-9 whitespace-separated entries:</P>
		<ol>
			<li>Probe sequence (must be in [acgtACGT], length must be in [1,27], 
			all probes must have same length).</li>
			<li>Alignment strand indicator: 1 or t or T or + if target (not 
			probe!) is on top strand, 0 or f or F or - otherwise</li>
			<li>Target sequence name</li>
			<li>Alignment position in target sequence (0-based, lower coordinate 
			of alignment)</li>
			<li>X coordinate of PM probe in array (0-based)</li>
			<li>Y coordinate of PM probe in array (0-based)</li>
		</ol>
		<P>For a PM Only sequence block:</P>
		<ol>
			<li value=7>Optional Match Score - float between 0.0 and 1.0</li>
		</ol>
		<p>For a PM and MM sequence block:</p>
		<ol>
			<li value=7>X coordinate of MM probe in array (0-based)</li>
			<li>Y coordinate of MM probe in array (0-based)</li>
			<li>Optional Match Score - float between 0.0 and 1.0</li>
		</ol>
		<p>Mixtures of PM only probes and PM/MM probe pairs are not supported 
			within a sequence block</p>
		<p>Tag/value support is added of the form:</p>
		<blockquote>
			<p>#tag value</p>
		</blockquote>
		<p>where tag and value are strings. The string naming the tag will not 
			include the leading '#'.
			Any lines with '#' alone as the first non-whitespace string are 
			treated as comments.</p>
		<p>If the Match Score is absent then it will be set to 1.0.</p>
		<p><span class="header4">Example</span><br>
		Below is a sample tpmap file:</p>
		<blockquote>
			<p><font face="Courier New" color="#000000">#seq_group_name HS1 (Required Line, whitespace delimited using 
			default std::cin, only first 2 fields used).<br>#version 11_Nov_2005 (Required Line, as above)<br>#tag1 value1 (Optional tag/value line, white space delimited, 
			leading # stripped from tagname)<br>#tag2 value2 (Another example of a tag/value optional line)<br>
			<br># this line is ignored because of white space after the #, empty 
			lines are ignored too<br><br>GCCCTGTTGTCTCTTACCCGGATGT f chr2 28 1112 2013 1112 2014<br>AATAGCCCTCATGTACGTCTCCTCC f chr2 1 1290 1449 1290 1450<br>AATAGCCCTCATGTACGTCTCCTCC f chr2 1 1291 1449 1291 1450<br>AATAGCCCTCATGTACGTCTCCTCC f chr2 1 1292 1449 1292 1450<br>GGAGGAGACGTACATGAGGGCTATT t chr2 1 1466 949 1466 950<br>
			<br># this sequence block inherits the group_name and version and other 
			tags<br>GTAATGGAGGGTAAGTTGAGAGACA t chr1 107 1729 1497 1729 1498<br>GGTAATGGAGGGTAAGTTGAGAGAC t chr1 108 509 1397 509 1398<br>TAGGGCTGTGTTAGGGTAGTGTTAG t chr1 64 1745 1095 1745 1096<br>GGTAATGGAGGGTAAGTTGAGAGAC t chr1 108 510 1397 510 1398<br>CACTACCCTAACACAGCCCTAATCT f chr1 68 991 1953 991 1954<br>GGTTAGATTAGGGCTGTGTTAGGGT t chr1 72 295 1987 295 1988<br>GTCTCTCAACTTACCCTCCATTACC f chr1 108 355 1437 355 1438<br>GTCTCTCAACTTACCCTCCATTACC F chr1 108 354 1437 354 1438<br>
			<br># a PMOnly block with a match score in line 2<br>GTAGAGAGATGGATGGTGGTTGGGA t chr3 474 2305 1565<br>TAAGTAGAGAGATGGATGGTGGTTG 1 chr3 477 1297 981 .9<br>TAAGTAGAGAGATGGATGGTGGTTG t chr3 477 1298 981<br>AGTAAGTAGAGAGATGGATGGTGGT t chr3 479 631 779<br>TAGTAAGTAGAGAGATGGATGGTGG t chr3 480 843 1211<br>TAGTAAGTAGAGAGATGGATGGTGG t chr3 480 844 1211<br>TAGTAAGTAGAGAGATGGATGGTGG t chr3 480 845 1211<br>TAGTAAGTAGAGAGATGGATGGTGG t chr3 480 846 1211<br>GTAGTAAGTAGAGAGATGGATGGTG t chr3 481 1397 1011<br>GTTGGTGGTAGTAAGTAGAGAGATG t chr3 488 2520 1035<br>GACGGTGGGTTGGTGGTAGTAAGTA t chr3 496 1000 2453<br>
			<br>#seq_group_name B_subtilis<br>#version 1/1/00:11am<br>#tag1 controls<br># has a match score in first line<br>CACACCCTAACACTACCCTAACACT 0 chr1 356 1069 1943 3069 1944 0.8<br>ACCCTAACACTACCCTAACACTACC f chr1 359 2059 2371 2059 2372<br>GGGTAGTGTTAGGGTAGTGTTAGGG t chr1 360 1704 2249 1704 2250<br>TAGGGTAGTGTTAGGGTAGTGTTAG T chr1 2012000111 1996 2209 1996 2210<br>
			<br><br># ###### Explanation of the sequence blocks:<br># There is a row corresponding to every instance of a (PM,MM) probe 
			pair or a PM probe aligning<br># to a target sequence. Each row consists of the following 6-9 
			whitespace-separated entries:<br><br># 1 Probe sequence (must be in [acgtACGT], length must be in [1,27], 
			all probes must have same length)<br># 2 Alignment strand indicator: 1 or t or T if target (not probe!) 
			is on top strand, 0 or f or F otherwise<br># 3 Target sequence name<br># 4 Alignment position in target sequence (0-based, lower coordinate 
			of alignment)<br># 5 X coordinate of PM probe in array (0-based)<br># 6 Y coordinate of PM probe in array (0-based)<br># For a PM Only sequence block:<br># 7 Optional Match Score - float between 0.0 and 1.0, represents how 
			well seq matches target<br># For a PM and MM sequence block:<br># 7 X coordinate of MM probe in array (0-based)<br># 8 Y coordinate of MM probe in array (0-based)<br># 9 Optional Match Score - float between 0.0 and 1.0<br># 
			<br># Mixtures of PM only probes and PM/MM probe pairs are not supported 
			within a sequence block</font></p>
		</blockquote>
	</td>
	</tr>
	



</table></TD></TR></TBODY>
</TABLE></TD></TR></TBODY>
</TABLE></TD></TR></TBODY>
</TABLE><p><BR> 
</p>
</BODY></HTML>